src/vm/eetwain.cpp

```diff
@@ -3805,6 +3806,28 @@ bool UnwindEbpDoubleAlignFrame(
     {
         TADDR baseSP;
 
+#ifdef WIN64EXCEPTIONS
+        // Funclets' frame pointers(EBP) are always restored so they can access to main function's local variables.
+        // Therefore the value of EBP is invalid for unwinder so we should use ESP instead.
+        // If funclet frame layout is changed from CodeGen::genFuncletProlog() and genFuncletEpilog(),
+        // we need to change here accordingly.
+        if (pCodeInfo->IsFunclet())
+        {
+            baseSP = curESP + 8; // padding for 16byte stack alignment allocated in genFuncletProlog()
```

Comment:
@BruceForstall Thank you for pointing it out! The current problem is that we do not have a frame pointer for this funclet frame (``ebp`` should be restored as that of main function as I understand). 

Could you let me know if there is any way to reserve any register (except ebp)? It would resolve issues from incorrect ESP if we could reserve any registers for a frame pointer for funclet frames.